algorithms/kernel/svm/oneapi/svm_train_thunder_oneapi_impl.i

```diff
@@ -0,0 +1,277 @@
+*/
+
+#ifndef __SVM_TRAIN_THUNDER_ONEAPI_IMPL_I__
+#define __SVM_TRAIN_THUNDER_ONEAPI_IMPL_I__
+
+#include "service/kernel/service_utils.h"
+#include "service/kernel/service_data_utils.h"
+#include "service/kernel/oneapi/blas_gpu.h"
+#include "externals/service_memory.h"
+#include "externals/service_ittnotify.h"
+#include "externals/service_service.h"
+#include "algorithms/kernel/svm/oneapi/cl_kernels/svm_train_block_smo_oneapi.cl"
+
+#include "algorithms/kernel/svm/oneapi/svm_train_cache_oneapi.h"
+#include "algorithms/kernel/svm/oneapi/svm_train_workset_oneapi.h"
+#include "algorithms/kernel/svm/oneapi/svm_train_result_oneapi.h"
+
+DAAL_ITTNOTIFY_DOMAIN(svm_train.default.batch);
+
+namespace daal
+{
+namespace algorithms
+{
+namespace svm
+{
+namespace training
+{
+namespace internal
+{
+using namespace daal::internal;
+using namespace daal::services::internal;
+using namespace daal::oneapi::internal;
+
+template <typename algorithmFPType, typename ParameterType>
+services::Status SVMTrainOneAPI<algorithmFPType, ParameterType, thunder>::updateGrad(const services::Buffer<algorithmFPType> & kernelWS,
+                                                                                     const services::Buffer<algorithmFPType> & deltaalpha,
+                                                                                     services::Buffer<algorithmFPType> & grad, const size_t nVectors,
+                                                                                     const size_t nWS)
+{
+    DAAL_ITTNOTIFY_SCOPED_TASK(updateGrad);
+    return BlasGpu<algorithmFPType>::xgemm(math::Layout::RowMajor, math::Transpose::Trans, math::Transpose::NoTrans, nVectors, 1, nWS,
+                                           algorithmFPType(1), kernelWS, nVectors, 0, deltaalpha, 1, 0, algorithmFPType(1), grad, 1, 0);
+}
+
+template <typename algorithmFPType, typename ParameterType>
+services::Status SVMTrainOneAPI<algorithmFPType, ParameterType, thunder>::smoKernel(
+    const services::Buffer<algorithmFPType> & y, const services::Buffer<algorithmFPType> & kernelWsRows, const services::Buffer<uint32_t> & wsIndices,
+    const uint32_t ldK, const services::Buffer<algorithmFPType> & f, const algorithmFPType C, const algorithmFPType eps, const algorithmFPType tau,
+    const uint32_t maxInnerIteration, services::Buffer<algorithmFPType> & alpha, services::Buffer<algorithmFPType> & deltaalpha,
+    services::Buffer<algorithmFPType> & resinfo, const size_t nWS)
+{
+    DAAL_ITTNOTIFY_SCOPED_TASK(smoKernel);
+
+    auto & context = services::Environment::getInstance()->getDefaultExecutionContext();
+    auto & factory = context.getClKernelFactory();
+
+    services::String build_options = getKeyFPType<algorithmFPType>();
+
+    services::String cachekey("__daal_algorithms_svm_smo_block_");
+    build_options.add(" -D WS_SIZE=");
+    char bufferString[DAAL_MAX_STRING_SIZE] = { 0 };
+    services::daal_int_to_string(bufferString, DAAL_MAX_STRING_SIZE, int(nWS));
```

Comment:
Is there ready to use function in DAAL similar to `std::stoi`?

Reply:
I did not find anything other than this function `services::daal_int_to_string` 😟